### PR TITLE
SR-11720: Thread.name leads to empty String on Linux

### PR DESCRIPTION
**Pull request by spevans**
- Check the return value of _CFThreadGetName correctly

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci test linux

**Comment by spevans** [comment]
@swift-ci test linux

**Comment by spevans** [comment]
@swift-ci test linux

File: Tests/Foundation/Tests/TestThread.swift [hunk 1]
```diff
@@ -64,57 +45,36 @@ class TestThread : XCTestCase {
...
-            condition.signal()
-            condition.unlock()
-        }
-        thread2.name = "Thread2-1"
-        thread2.start()
-
-        // Allow 1 second for thread2 to finish
-        XCTAssertTrue(condition.wait(until: Date(timeIntervalSinceNow: 1)))
-        condition.unlock()
+        testInternalThreadName("mainThread")
```
Comment (3405691582) [inline Tests/Foundation/Tests/TestThread.swift]: This might be something to address in another pr, but since thread names aren't supported on TARGET_OS_BSD (that is, `pthread_setname_np` nor `pthread_getname_np` are available), these tests will trip expectedly. Maybe the #if clauses can be tweaked?

Comment (spevans) [reply Tests/Foundation/Tests/TestThread.swift]: I think for targets that dont support `pthread_{get,set}name_np`, the name can just be stored in a `String` in the `Thread` object.

Do the `swift-corelibs-foundation` tests build & run at all on BSD and is this going to break them or is it just another thing to remember to fix when the tests are ported?

Comment (3405691582) [reply Tests/Foundation/Tests/TestThread.swift]: If it's as simple as a string variable in the other cases, then I think we can leave it as "another thing to remember to fix when the tests are ported". 

(There's still more hurdles to jump through on the porting -- like Dispatch and friends, _especially_ -- but with the WIP Swift BSD port, a decent percentage of Foundation tests appear to be running and passing.)

**Comment by spevans** [comment]
@compnerd Can you run a windows-ci build to check this wont break it, thanks
